gammapy/scripts/spectrum_pipe.py

```diff
@@ -73,30 +72,74 @@ def run_extraction(self):
 
         self.extraction.run()
 
+    @property
+    def _result_dict(self):
+        """Convert to dict."""
+        val = dict()
+        model = self.config["fit"]["model"]
+        val["model"] = model.to_dict()
+
+        fit_range = self.config["fit"].get("fit_range")
+
+        if fit_range is not None:
+            val["fit_range"] = dict(
+                min=fit_range[0].value,
+                max=fit_range[1].value,
+                unit=fit_range.unit.to_string("fits"),
+            )
+
+        val["statval"] = float(self.fit_result.total_stat)
+        val["statname"] = "wstat"
+
+        return val
+
+    def write(self, filename, mode="w"):
+        """Write to YAML file.
+
+        Parameters
+        ----------
+        filename : str
+            File to write
+        mode : str
+            Write mode
+        """
+        d = self._result_dict
+        val = yaml.safe_dump(d, default_flow_style=False)
+
+        with open(str(filename), mode) as outfile:
+            outfile.write(val)
+
     def run_fit(self, optimize_opts=None):
         """Run all step for the spectrum fit."""
-        self.fit = SpectrumFit(
-            obs_list=self.extraction.spectrum_observations, **self.config["fit"]
-        )
-        self.fit.run(optimize_opts=optimize_opts)
-        modelname = self.fit.result[0].model.__class__.__name__
+        datasets_fit = self.extraction.spectrum_observations.to_spectrum_datasets(**self.config["fit"])
```

Comment:
I suppose we can keep like this for the moment, but this will likely have to significantly change if `SpectrumObservation` and `SpectrumObservationList` are removed.

I suppose that a loop on all `Dataset` to set the model is unavoidable step for the user.

Reply:
Yes, this is only meant to be a preliminary solution to help with the transition from `SpectrumObservation` to `SpectrumDataset`. I've opened a reminder issue #2113 to discuss the API question how to set models on `Datasets`. Maybe keeping the loop over dataset objects is just fine...